app/javascript/src/NewService/index.jsx

```diff
@@ -0,0 +1,127 @@
+import 'core-js/es6/promise'
+
+import ServiceSourceForm from 'NewService/components/ServiceSourceForm'
+import ServiceDiscoveryForm from 'NewService/components/ServiceDiscoveryForm'
+import ServiceNewForm from 'NewService/components/ServiceNewForm'
+
+const BASE_URL = '/p/admin/service_discovery/'
+const NAMESPACES_URL = `${BASE_URL}projects.json`
+
+import type {SourceFormType, ServiceDiscoveryFormType, ServiceNewFormType} from 'NewService/types'
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+const fetchData = async (url: string) => {
+  const dataItem = url === NAMESPACES_URL
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url).then(response => {
+    if (!response.ok) {
+      throw Error(response.statusText)
+    }
+    return response.json()
+  })
+    .then(data => data[dataItem].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => console.error(error))
+}
+
+// eslint-disable max-lines-per-function
+const NewServiceWrapper = (props: Props) => {
+  const [isServDescVisible, setServDescVisible] = useState(false)
+  const [isServNewVisible, setServNewVisible] = useState(true)
+  const [namespaces, setNamespaces] = useState([])
+  const [services, setServices] = useState([])
+  const [selectedNamespace, setSelectedNamespace] = useState(undefined)
+  const [servicesRequested, setServicesRequested] = useState(false)
+
+  const settingSelectedNamespace = (namespace: string) => {
+    setSelectedNamespace(namespace)
+  }
+
+  const onChangeNamespaces = (event: SyntheticEvent<HTMLSelectElement>) => {
+    setSelectedNamespace(event.currentTarget.value)
+    setServicesRequested(true)
+  }
+  // eslint-disable-next-line flowtype/no-weak-types
+  const startNamespacesFetch = async (): Promise<any> => {
+    const namespacesData = await fetchData(NAMESPACES_URL)
+    if (namespacesData) {
+      setNamespaces(namespacesData)
+      setServicesRequested(true)
+      settingSelectedNamespace(namespacesData[0])
+    }
+  }
+
+  const startServicesFetch = async (servicesUrl: string) => {
+    setServices([])
+    const servicesData = await fetchData(servicesUrl)
+    if (servicesData) {
+      setServices(servicesData)
+      setServicesRequested(false)
+    }
+  }
+
+  const handleFormsVisibility = (event: SyntheticEvent<HTMLInputElement>) => {
+    if (event.currentTarget.value === 'discover') {
+      setServDescVisible(true)
+      setServNewVisible(false)
+    } else {
+      setServDescVisible(false)
+      setServNewVisible(true)
+    }
+  }
+
+  useEffect(() => {
+    if (servicesRequested && selectedNamespace) {
+      const SERVICES_URL = `${BASE_URL}namespaces/${selectedNamespace}/services.json`
+      startServicesFetch(SERVICES_URL)
+    }
+  }, [servicesRequested, selectedNamespace])
+
+  const formsProps: {
```

Comment:
Watch out for this mega props objects, it's redundant. You can simply define a different const for each component.
I would set props directly to the component, without the spread operator.